aws/retry/middleware_test.go

```diff
@@ -153,6 +155,9 @@ func TestAttemptMiddleware(t *testing.T) {
 					MaxAttempts: 3,
 				},
 			},
+			ExpectResults: AttemptResults{Results: []AttemptResult{
+				{},
```

Comment:
even for a single attempt, no error, shouldn't an `AttemptResult` be present with at minimum its Response member populated?

Though it looks like the nested metadata item covers that a little. Maybe have test where the `next` does have a `Response` value, and on that doesn't.